gio/src/subclass/vfs.rs

```diff
@@ -0,0 +1,304 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use glib::{prelude::*, subclass::prelude::*, translate::*, GString};
+
+use libc::c_char;
+
+use crate::{ffi, File, Vfs};
+
+pub trait VfsImpl: ObjectImpl + ObjectSubclass<Type: IsA<Vfs>> {
+    fn is_active(&self) -> bool {
+        self.parent_is_active()
+    }
+
+    fn get_file_for_path(&self, path: &GString) -> File {
```

Comment:
You probably also need to implement the required pieces to implement `gio::File` for this to be useful, or not?